pkg/cloud/gcp/actuators/util/gcp_tags_labels_test.go

```diff
@@ -14,20 +14,505 @@ limitations under the License.
+			Key:      "key23",
+			Value:    "value23"},
+		{
+			ParentID: "openshift",
+			Key:      "key24",
+			Value:    "value24"},
+		{
+			ParentID: "openshift",
+			Key:      "key25",
+			Value:    "value25"},
+		{
+			ParentID: "openshift",
+			Key:      "key26",
+			Value:    "value26"},
+		{
+			ParentID: "openshift",
+			Key:      "key27",
+			Value:    "value27"},
+		{
+			ParentID: "openshift",
+			Key:      "key28",
+			Value:    "value28"},
+		{
+			ParentID: "openshift",
+			Key:      "key29",
+			Value:    "value29"},
+		{
+			ParentID: "openshift",
+			Key:      "key30",
+			Value:    "value30"},
+		{
+			ParentID: "openshift",
+			Key:      "key31",
+			Value:    "value31"},
+		{
+			ParentID: "openshift",
+			Key:      "key32",
+			Value:    "value32"},
+		{
+			ParentID: "openshift",
+			Key:      "key33",
+			Value:    "value33"},
+		{
+			ParentID: "openshift",
+			Key:      "key34",
+			Value:    "value34"},
+		{
+			ParentID: "openshift",
+			Key:      "key35",
+			Value:    "value35"},
+		{
+			ParentID: "openshift",
+			Key:      "key36",
+			Value:    "value36"},
+		{
+			ParentID: "openshift",
+			Key:      "key37",
+			Value:    "value37"},
+		{
+			ParentID: "openshift",
+			Key:      "key38",
+			Value:    "value38"},
+		{
+			ParentID: "openshift",
+			Key:      "key39",
+			Value:    "value39"},
+		{
+			ParentID: "openshift",
+			Key:      "key40",
+			Value:    "value40"},
+		{
+			ParentID: "openshift",
+			Key:      "key41",
+			Value:    "value41"},
+		{
+			ParentID: "openshift",
+			Key:      "key42",
+			Value:    "value42"},
+		{
+			ParentID: "openshift",
+			Key:      "key43",
+			Value:    "value43"},
+		{
+			ParentID: "openshift",
+			Key:      "key44",
+			Value:    "value44"},
+		{
+			ParentID: "openshift",
+			Key:      "key45",
+			Value:    "value45"},
+		{
+			ParentID: "openshift",
+			Key:      "key46",
+			Value:    "value46"},
+		{
+			ParentID: "openshift",
+			Key:      "key47",
+			Value:    "value47"},
+		{
+			ParentID: "openshift",
+			Key:      "key48",
+			Value:    "value48"},
+		{
+			ParentID: "openshift",
+			Key:      "key49",
+			Value:    "value49"},
+		{
+			ParentID: "openshift",
+			Key:      "key50",
+			Value:    "value50"},
+		{
+			ParentID: "openshift",
+			Key:      "key51",
+			Value:    "value51"},
+		{
+			ParentID: "openshift",
+			Key:      "key52",
+			Value:    "value52"},
+		{
+			ParentID: "openshift",
+			Key:      "key101",
+			Value:    "value101"},
+		{
+			ParentID: "openshift",
+			Key:      "key501",
+			Value:    "value501"},
+	}
+)
+
+// tagSetIndexRange is for picking defined tags in the given index range
+// from testTagSets. low and high are both inclusive bounds.
+type tagSetIndexRange struct {
+	low  int
+	high int
+}
+
+// getTestTagSet returns the list of tags in the provided multiple ranges from the
+// testTagSets.
+func getTestTagSet[T configv1.GCPResourceTag | machinev1.ResourceManagerTag](tags *[]T, indexRange ...tagSetIndexRange) {
```

Comment:
Yeah, both are structs having the same exact member fields, and since both are populated from the common source, thought to use generics and cast it to the requested type.